### PR TITLE
chore: change top level folder structure

### PR DESCRIPTION
**Pull request by jsonsivar**
Some notes to help the review since this is a large PR but a small change (if that makes sense).

- Created top level folder for contracts and moved all contracts in there
- Split the package.json and moved contract related things into contracts/package.json but kept a root package.json for husky and other things we may need
- Updated various references to folders in husky, workflows, readmes etc.

Work being done in UI and subgraph should mostly be unaffected and merge should be smooth. For contract work we should try to branch of this then maybe add the commits on top.

### CODE REVIEW

File: contracts/README.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+```
+yarn deploy:mainnet
+```
+
+to deploy the contract on the testnet. Please note that your wallet needs to hold enough Mumbai MATIC for the deployment to be successful.
+-->
+
+## ▶️ Interaction scripts
+
+Right away, in the [/scripts](/scripts/) folder you are able to see some scripts that will help you to interact with deployed contracts. By default you are able to select `localhost`, `hardhat` or `mumbai` network name predefined on [hardhat.config.ts](/hardhat.config.ts). The scripts will be using the deployment information stored in the [/deployments](/deployments/) folder. You should have a nested folder for each of the networks you have deployed it. The scripts needs be run using the Hardhat environment following the pattern:
```
Comment (zoruka) [inline contracts/README.md:175]:
```suggestion
Right away, in the [/scripts](./scripts/) folder you are able to see some scripts that will help you to interact with deployed contracts. By default you are able to select `localhost`, `hardhat` or `mumbai` network name predefined on [hardhat.config.ts](./hardhat.config.ts). The scripts will be using the deployment information stored in the [/deployments](./deployments/) folder. You should have a nested folder for each of the networks you have deployed it. The scripts needs be run using the Hardhat environment following the pattern:
```

File: package.json [hunk 2]
```diff
@@ -29,32 +21,15 @@
...
-    "@nomicfoundation/hardhat-network-helpers": "^1.0.7",
-    "@nomicfoundation/hardhat-toolbox": "^2.0.0",
-    "@nomiclabs/hardhat-ethers": "^2.2.1",
-    "@nomiclabs/hardhat-etherscan": "^3.1.0",
-    "@nomiclabs/hardhat-web3": "^2.0.0",
-    "@openzeppelin/contracts": "^4.7.3",
-    "@openzeppelin/contracts-upgradeable": "^4.8.0",
-    "@openzeppelin/hardhat-upgrades": "^1.22.0",
     "@types/mocha": "^10.0.1",
     "chai": "^4.3.6",
```
Comment (zoruka) [inline package.json:25]: These libraries for testing with Hardhat
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -29,32 +21,15 @@
...
     "lint-staged": "^13.0.4",
     "pinst": "^3.0.0",
     "prettier": "^2.7.1",
     "prettier-plugin-solidity": "^1.0.0",
-    "sharp": "^0.31.3",
-    "solidity-coverage": "^0.8.2",
     "ts-node": "^10.9.1",
-    "typescript": "^4.9.3",
-    "web3": "^1.8.1"
+    "typescript": "^4.9.3"
```
Comment (zoruka) [inline package.json:33]: I'm not 100% sure but I think the only packages needed in the `package.json` are `husky`, `pinst` and `prettier`. Maybe we could force `prettier` from underneath projects to use the config from root folder.

File: .gitignore [hunk 4]
```diff
@@ -1,24 +1,3 @@
 node_modules
 .env
-
```
Comment (EmperorOrokuSaki) [inline .gitignore:3]: question: does having multiple .gitignore files in one repository work? I thought only the one that is in the root folder is considered by Git.

File: README.md [hunk 5]
```diff
@@ -21,197 +21,23 @@ You can find the wiki [here](https://github.com/fleekxyz/non-fungible-apps/wiki)
...
 
 Inside the root folder you are going to find:
-
 - [/contracts](/contracts): all the developed contracts
-- [/deploy](/deploy): scripts used for deployment on different networks
-- [/deployments](/deployments): resultant ABI and info for deployments (each network will have a nested folder)
-- [/lib](/lib): external modules used by Foundry
-- [/scripts](/scripts): any utility scripts used for interacting with deployed contracts
-- [/test](/test): tests suits to validate contracts
+- [/subgraph](/contracts): The Graph project related code
```
Comment (EmperorOrokuSaki) [inline README.md:25]:
```suggestion
- [/subgraph](/subgraph): The Graph project related code

```

**Review by EmperorOrokuSaki — COMMENTED** [review]
This looks good to me after Zoruka's comments and the one suggestion I made are committed.

File: .gitignore (hunk 4 above)
Comment (jsonsivar) [reply .gitignore:3]: Yeah it works, and was suggested by @zoruka in the previous PR. 
image

Looked it up and it says that so we just need to be careful of how they are applied together but it works. That's why those files didn't get committed either.

File: package.json (hunk 3 above)
Comment (jsonsivar) [reply package.json:33]: Agreed 100%

**Review by zoruka — APPROVED** [review]

**Pull request merged**